packages/core/http/core-http-server/src/router/route_validator.ts

```diff
@@ -156,3 +183,14 @@ export interface RouteValidatorRequestAndResponses<P, Q, B> {
 export type RouteValidator<P, Q, B> =
   | RouteValidatorFullConfigRequest<P, Q, B>
   | RouteValidatorRequestAndResponses<P, Q, B>;
+
+/**
+ * A validation schema factory.
+ *
+ * @note Often used to lazily create schemas that are otherwise not needed
+ * @note assume that these factory will only be called once
+ *
+ * @return A @kbn/config-schema schema
+ * @public
+ */
+export type LazyValidator = () => ObjectType | Type<unknown>;
```

Comment:
NIT: why not just only `() => Type<unknown>` given `ObjectType` already extends `Type<Something>`?

(I see this is coming from the diff line 137->160, but the question could have been the same for before your change)